pennylane/operation.py

```diff
@@ -1507,7 +1508,7 @@ def data(self):
         Returns:
             list[Any]: flattened list containing all dependent parameters
         """
-        return [p for sublist in [o.data for o in self.obs] for p in sublist]
+        return sum((o.data for o in self.obs), [])
```

Comment:
I must admit, I'm having a bit of trouble parsing what is happening here! Is this equivalent to?
```python
[] + [self.obs[0].data, self.obs[1].data, ...]
```

Reply:
Ah, yes, exactly. This is because the initial value of Python's `sum` defaults to `0` ,so we need to give an explicitly different value, namely a list.
I would prefer to provide it via `sum(..., start=[])` but unfortunately, passing the `start` value via a named kwarg only is possible since Python 3.8

Reply:
Sorry @josh146 I wasn't reading carefully enough. It is actually equivalent to
```python
reduce(list.__add__, [o.data for o in self.obs])
```
or as pseudocode
```python
[] + self.obs[0].data + self.obs[1].data + ...
```
The difference is the flat structure of the resulting list.

Reply:
oh of course!